### PR TITLE
Update xray_watch resource API

### LINKED ISSUE #82: API for adding resources to watches isn't intuitive
Currently the API for adding a repository (or any type of resource) is like so:

```terraform
resource "artifactory_xray_watch" "test" {
  name        = "watch-npm-local-repo"
  description = "apply a severity-based policy to the npm local repo"

  resources {
    type       = "repository"
    name       = artifactory_local_repository.npm-local.key
    bin_mgr_id = "example-com-artifactory-instance"
    repo_type  = "local"
  }
}
```

I strongly believe that the `resources` key should be singular (i.e. `resource`) since it is intended to be specified once per resource you intend to add to the xray watch. This would make this more consistent with the core terraform modules.

### PR DESCRIPTION
**Pull request by jasonwbarnett**
This fixes #82

This was fully tested and works like a beauty. This will automatically upgrade the user's state file and after doing so it will prevent them from running that state against an older version of the provider.

### CODE REVIEW

**Comment by chb0github** [comment]
Can you please update the example.tf that makes use of this? I see no tests updated/created either

**Comment by jasonwbarnett** [comment]
@chb0github I updated all docs and tests, etc.

File: sample.tf [hunk 1]
```diff
@@ -68,7 +68,7 @@ resource "artifactory_xray_watch" "test" {
   name        = "watch-npm-local-repo"
   description = "apply a severity-based policy to the npm local repo"
 
-  resources {
+  resource {
```
Comment (chb0github) [inline sample.tf:71]: Why the change from plural?

**Comment by chb0github** [comment]
I don't understand the change _from_ plural - changing that changes how the TF file behaves and we have to maintain backward compatibility

Am I missing something?

**Comment by jasonwbarnett** [comment]
The actual API is illogical for starters. You specify the resource multiple time and the resource block only contains a single resource. Another reason is it aligns with the actual core providers that Hashicorp and others have built. See the examples below:

- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#admin_ssh_key
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#secret
- https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#attached_disk
- https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#network_interface

You'll notice the pattern is that their names are singular and are able to have multiple occurrences of the argument.

So in practice it doesn't change anything with how the provider behaves. It's just a simple API change that is more intuitive. There really shouldn't be a concern with backwards compatibility because there is a state file migrator and the reads will also update the sate file, even if the state file wasn't migrated (although it will be).

So the change should be completely seamless with the exception that as people upgrade to the version this would be released in they would need to update their terraform from the plural (i.e. `resources`) to singular (i.e. `resource`).

**Comment by chb0github** [comment]
I can't merge this until I have time to take a deeper think on impact - if it breaks people, we generally frown upon that. A major version bump might be good enough

**Comment by chb0github** [comment]
@jasonwbarnett what do you think of #96

**Comment by jasonwbarnett** [comment]
> @jasonwbarnett what do you think of #96 

I'll peak at this over the weekend and give my thoughts directly on the PR.

File: docs/resources/artifactory_xray_policy.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ resource "xray_policy" "example" {
 resource "xray_watch" "example" {
   name  = "watch-name"
   description = "watching all repositories"
-  resources {
+  resource {
```
Comment (chb0github) [inline docs/resources/artifactory_xray_policy.md:27]: Let's go back to our pluralized form

**Comment by chb0github** [comment]
Closing this per conversation
